oodeel/methods/base.py

```diff
@@ -190,32 +191,40 @@ def score(
         dataset: Union[ItemType, DatasetType],
     ) -> np.ndarray:
         """
-        Computes an OOD score for input samples "inputs"
+        Computes an OOD score for input samples "inputs".
 
         Args:
             dataset (Union[ItemType, DatasetType]): dataset or tensors to score
 
         Returns:
-            scores or list of scores (depending on the input)
+            tuple: scores or list of scores (depending on the input) and a dictionary
+                containing logits and labels.
         """
         assert self.feature_extractor is not None, "Call .fit() before .score()"
 
         # Case 1: dataset is neither a tf.data.Dataset nor a torch.DataLoader
         if isinstance(dataset, get_args(ItemType)):
             tensor = self.data_handler.get_input_from_dataset_item(dataset)
-            scores = self._score_tensor(tensor)
+            scores, info = self._score_tensor(tensor)
         # Case 2: dataset is a tf.data.Dataset or a torch.DataLoader
         elif isinstance(dataset, get_args(DatasetType)):
             scores = np.array([])
+            info = None
             for tensor in dataset:
                 tensor = self.data_handler.get_input_from_dataset_item(tensor)
-                score_batch = self._score_tensor(tensor)
+                score_batch, info_batch = self._score_tensor(tensor)
                 scores = np.append(scores, score_batch)
+                if info is None:
+                    info = info_batch
+                else:
+                    for key in info.keys():
+                        if info_batch[key] is not None:
+                            info[key] = self.op.cat([info[key], info_batch[key]])
```

Comment:
I may be wrong but it seems to me that if the input tensor only is provided to `self._score_tensor` and that the info is returned from this method, we will never have access to the labels in the info? The `_score_tensor` methods access to the feature extractor predictions through `features, info = self.feature_extractor.predict(inputs)`, but inputs do not contain the labels so info dict returns None for the label key... 

I think that `_score_tensor` should instead access to the feature extractor predictions through `features = self.feature_extractor.predict_tensor(inputs)` and return a tuple `(scores, logits)`. Then the info dictionary should be built in the `score` function which has access to the labels!